### PR TITLE
feat: disable umami analytics on localhost for dashboard and classroomio-com

### LINKED ISSUE #493: Disable Umami analytics on localhost
### Feature Description

This affects:
- apps/classroomio-com
- apps/dashboard

### Use Case

_No response_

### Proposed Solution

Here is a way to do it: https://github.com/umami-software/umami/discussions/852

### Alternatives (optional)

_No response_

### Additional Context

_No response_

### Please check the boxes that apply to this feature request.

- [ ] I have searched the existing feature requests to make sure this is not a duplicate.
- [ ] I have provided a detailed description of the requested feature.
- [ ] I have explained the use case or scenario for this feature.
- [ ] I have included any relevant technical details or design suggestions.
- [ ] I understand that this is a suggestion and that there is no guarantee of implementation.

### PR DESCRIPTION
**Pull request by codiini**
## What does this PR do?

Fixes #493

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [x] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change adds a new database migration
- [ ] This change requires a documentation update

## How should this be tested?

- Check Umami Dashboard if localhost interactions on Landing page and dashboard are being recorded.

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [ ] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the ClassroomIO Docs if changes were necessary

### CODE REVIEW

File: apps/classroomio-com/src/app.html [hunk 1]
```diff
@@ -98,6 +98,11 @@
       layout: 'month_view'
     });
   </script>
+
+  <script type="text/javascript">
+    window.localStorage.setItem('umami.disabled', 1)
+  </script>
```
Comment (rotimi-best) [inline apps/classroomio-com/src/app.html]: @codiini Shouldn't we be doing this only if the environment is localhost?

Won't this disable umami completely?

File: apps/dashboard/src/app.html [hunk 2]
```diff
@@ -22,6 +22,10 @@
 
   <script src="https://cdn.plyr.io/3.7.3/plyr.js"></script>
 
+  <script type="text/javascript">
+    window.localStorage.setItem('umami.disabled', 1)
+  </script>
```
Comment (rotimi-best) [inline apps/dashboard/src/app.html]: Same thing here.

File: apps/classroomio-com/src/app.html (hunk 1 above)
Comment (rotimi-best) [reply apps/classroomio-com/src/app.html]: From the docs here, it completely stops umami from collecting analytics https://umami.is/docs/guides/exclude-my-own-visits

I think what we should be doing is updating `src/routes/+layout.svelte`

```ts
import { dev } from '$app/environment';
import { onMount } from 'svelte';

onMount(() => {
  if (dev) {
    localStorage.setItem('umami.disabled', 1)
  }
})
```

**Review by rotimi-best — COMMENTED** [review]
Thanks for the PR 😉.

Please take a look at my comments

File: apps/classroomio-com/src/app.html (hunk 1 above)
Comment (codiini) [reply apps/classroomio-com/src/app.html]: Oh yes, my bad. I ready the linked Discussion on the PR and skipped straight to the solution without looking at further context. Fixing now.

**Comment by codiini** [comment]
@rotimi-best Issue resolved

**Review by rotimi-best — APPROVED** [review]
LGTM 🙏🏽🥳

**Pull request merged**